### PR TITLE
Randomize optionals and minItems + clean up

### PR DESCRIPTION
**Pull request by idantene**
- Use the `s/seedHack/runnerHack/g` to include more options for randomization and generation.
- Specifically, increase `minItems` by a randomized, exponential increase, and add `optionalsProbability` (randomized at each run).
- Remove old serializers and deserializers which weren't in use anyway.
- Remove `service/constants.ts`, `service/constants.ts` and `backend/constants.ts` which weren't in use anyway.
- Remove `XRegExp` from dependency (should we have a sweep over all dependencies in general?)

### CODE REVIEW

File: packages/unmock-core/src/generator.ts [hunk 1]
```diff
@@ -52,8 +52,15 @@ import {
   Schema,
 } from "./service/interfaces";
 import { ServiceStore } from "./service/serviceStore";
-export const seedHack = {
+export const runnerHack = {
```
Comment (ksaaskil) [inline packages/unmock-core/src/generator.ts]: I would maybe just name these as "runnerConfiguration" or something more sensible 🙂

File: packages/unmock-core/src/generator.ts [hunk 2]
```diff
@@ -52,8 +52,15 @@ import {
...
   seed: 0,
+  optionalsProbability: 1.0,
+  minItems: 0,
+};
+export const resetRunnerHack = () => {
```
Comment (ksaaskil) [inline packages/unmock-core/src/generator.ts]: Something like `runner.configuration = DEFAULT_CONFIGURATION` would maybe reduce the need to remember to define these in two places.

Comment (idantene) [reply packages/unmock-core/src/generator.ts]: It's actually meant to be used within the generator, to reset `minItems`.
I can't decide if it's better after generation or after `runner` is complete. But TBH I don't understand yet how `runner` works, so I preferred putting it here :)
(see line num 727)

Comment (idantene) [reply packages/unmock-core/src/generator.ts]: Now that I do, I'll just move it there.

**Pull request merged**